lesson_7/lesson7_decorators.py

```diff
@@ -0,0 +1,65 @@
+# "GREETINGS WORLD!"
+
+def uppercase(fnc):
+    def inner_fnc(*args):
+        initial_str = fnc(*args)
+        uppercase_str = initial_str.upper()
+        return uppercase_str
+
+    return inner_fnc
+
+
+@uppercase
+def greet(name):
+    return "Greetings {}!".format(name)
+
+
+print(greet('world'))
+
+
+# Create a decorator called safe_divide that will output a message if the division cannot be performed, othervise it will return the result.
+
+def safe_divide(fnc):
+
+   def wrapper(*args, **kwargs):
+        try:
+            return fnc(*args, **kwargs)
+        except Exception as e:
+            print("Division by zero error")
```

Comment:
Would this still be a valid error if I try the following:
```
divide(10, 'd')
```
?

Reply:
changed the error message to "Division error"